core/src/autogluon/core/calibrate/temperature_scaling.py

```diff
@@ -47,18 +47,26 @@ def tune_temperature_scaling(y_val_probs: np.ndarray, y_val: np.ndarray, init_va
     optimizer = torch.optim.LBFGS([temperature_param], lr=lr, max_iter=max_iter)
     scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=0.99)
 
+    optimizer_trajectory = []
+
     def temperature_scale_step():
         optimizer.zero_grad()
         temp = temperature_param.unsqueeze(1).expand(logits.size(0), logits.size(1))
         new_logits = logits / temp
         loss = nll_criterion(new_logits, y_val_tensor)
         loss.backward()
         scheduler.step()
+        optimizer_trajectory.append((loss.item(), temperature_param.item()))
```

Comment:
Is this the correct location for this line? Or should it be before the `loss.backward()` call?

Reply:
I think it does not matter, as `backward` should only affect the connected computation graph, not the loss value stored in the tensor. In [this ](https://pytorch.org/tutorials/beginner/introyt/trainingyt.html#the-training-loop) example code, they also call `loss.item()` after `loss.backward()`.